### PR TITLE
Add visually-hidden text to headers of ComparisonTable feature columns

### PR DESCRIPTION
**Pull request by joshfarrant**
## List of notable changes:

- Featured ComparisonTable columns now feature the visually-hidden text `(featured)` after the column name 

## What should reviewers focus on?

- Check that this is read correctly by screen readers

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/3752

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/ComparisonTable/ComparisonTable.tsx
Comment (rezrah) [inline packages/react/src/ComparisonTable/ComparisonTable.tsx:76]: qq. is there a way we can make this customizable by end users? This is currently a hard coded English string, which we'll need to eventually change for localization reasons, but there's also an edge case where content may need to change to better describe the table cell header more accurately (feature / highlighted / spotlighted). I think the the flexibility would be nice to offer on the API.. Not something we need to solve now if it balloons the scope though.

**Review by rezrah — APPROVED** [review]
Works as expected to resolve the Issue.

Highlighted one item that we'll need to come back to at a later date, but we can defer to a follow up. 

Thanks for fixing this @joshfarrant

**Comment by joshfarrant** [comment]
@rezrah Great shout, we could just add a `featuredColumnAccessibleLabel` required prop that defaults to `featured` which we always render inside a pair of brackets. wdyt?

**Comment by joshfarrant** [comment]
I've added that @rezrah, let me know your thoughts

**Comment by rezrah** [comment]
> @rezrah Great shout, we could just add a `featuredColumnAccessibleLabel` required prop that defaults to `(featured)`. wdyt?

Yep sounds good, your name suggestion makes sense too, though wondering if we can shorten it, or make it more consistent with how we represent visually hidden content. 

Some other suggestions:

- `featuredLabel`
- `featuredColumnLabel`
- `visuallyHiddenFeaturedLabel` similar to [this](https://github.com/primer/brand/blob/main/packages/react/src/VideoPlayer/VideoPlayer.tsx#L31)

File: packages/react/src/ComparisonTable/ComparisonTable.tsx
Comment (rezrah) [inline packages/react/src/ComparisonTable/ComparisonTable.tsx:32]: Could we apply the parenthesis internally please, and not leave to end user to customize in case they forget?

Comment (joshfarrant) [reply packages/react/src/ComparisonTable/ComparisonTable.tsx]: 👍 794484bd7f9779e9bda0d2697580567c9ba4eeef

File: apps/docs/content/components/ComparisonTable.mdx
Comment (rezrah) [inline apps/docs/content/components/ComparisonTable.mdx:129]: FYI This isn't closed properly so is breaking docs. Guessing it's not needed?

```suggestion
```

Screenshot 2024-08-14 at 15 21 14

File: apps/docs/content/components/ComparisonTable.mdx
Comment (rezrah) [inline apps/docs/content/components/ComparisonTable.mdx:368]:
```suggestion
| `visuallyHiddenFeaturedLabel` | `string`                                                                                | `'featured'`  | `false`  | A visually-hidden label which is appended to the end of the column title. Used to identify the featured column to screen reader users.                    |
```

**Review by rezrah — APPROVED** [review]
Looks great, just two more changes before it's merged (one is super important)

File: apps/docs/content/components/ComparisonTable.mdx
Comment (joshfarrant) [reply apps/docs/content/components/ComparisonTable.mdx:129]: Ah sorry, Copilot has been really eager to add this every opportunity it gets. I keep deleting them but obviously I missed it this time 🤦‍♂️

File: apps/docs/content/components/ComparisonTable.mdx
Comment (joshfarrant) [reply apps/docs/content/components/ComparisonTable.mdx]: Good spot

**Pull request merged**
